docs/Solidity.g4

```diff
@@ -23,7 +23,7 @@ version
   : versionConstraint versionConstraint? ;
```

Comment:
Would getting rid of the `| expression` above be another solution?

Reply:
```
pragmaValue
  : version | expression ;
```
is a parser rule. The parser processes tokens which are already created by the lexer, and the lexer thinks that `0.5` is `DecimalNumber`, not `VersionLiteral`. So with or without `| expression`, the parser gets a wrong token.

The right thing seems to be changing the lexer and using [lexical modes](https://github.com/antlr/antlr4/blob/master/doc/lexer-rules.md), but that requires a separate lexer grammar.